Extensions/PrimitiveDrawing/shapepainterruntimeobject-pixi-renderer.ts

```diff
@@ -268,21 +301,165 @@ namespace gdjs {
       );
     }
 
-    updateXPosition(): void {
-      if (!this._object._absoluteCoordinates) {
-        this._graphics.position.x = this._object.x;
-      } else {
+    invalidateBounds() {
+      this._object.invalidateBounds();
+      this._positionXIsUpToDate = false;
+      this._positionYIsUpToDate = false;
+    }
+
+    updatePreRender(): void {
+      this.updatePositionIfNeeded();
+    }
+
+    updatePositionX(): void {
+      if (this._object._absoluteCoordinates) {
+        this._graphics.pivot.x = 0;
         this._graphics.position.x = 0;
+      } else {
+        // Make the drawing rotate around the anchor.
+        this._graphics.pivot.x = this._object.getRotationAnchorX();
+        // Multiply with the scale to have the scale anchor
+        // at the object position instead of the center.
+        this._graphics.position.x =
+          this._object.x +
+          this._graphics.pivot.x * Math.abs(this._graphics.scale.x);
       }
     }
 
-    updateYPosition(): void {
-      if (!this._object._absoluteCoordinates) {
-        this._graphics.position.y = this._object.y;
-      } else {
+    updatePositionY(): void {
+      if (this._object._absoluteCoordinates) {
+        this._graphics.pivot.y = 0;
         this._graphics.position.y = 0;
+      } else {
+        this._graphics.pivot.y = this._object.getRotationAnchorY();
+        this._graphics.position.y =
+          this._object.y +
+          this._graphics.pivot.y * Math.abs(this._graphics.scale.y);
+      }
+    }
+
+    updatePositionIfNeeded() {
+      if (!this._positionXIsUpToDate) {
+        this.updatePositionX();
+        this._positionXIsUpToDate = true;
+      }
+      if (!this._positionYIsUpToDate) {
+        this.updatePositionY();
+        this._positionYIsUpToDate = true;
       }
     }
+
+    updateAngle(): void {
+      if (this._object._absoluteCoordinates) {
+        this._graphics.angle = 0;
+      } else {
+        this._graphics.angle = this._object.angle;
+      }
+    }
+
+    updateScaleX(): void {
+      if (this._object._absoluteCoordinates) {
+        this._graphics.scale.x = 1;
+      } else {
+        this._graphics.scale.x = this._object._scaleX;
+      }
+      // updatePositionX() uses scale.x
+      this._positionXIsUpToDate = false;
+    }
+
+    updateScaleY(): void {
+      if (this._object._absoluteCoordinates) {
+        this._graphics.scale.y = 1;
+      } else {
+        this._graphics.scale.y = this._object._scaleY;
+      }
+      // updatePositionY() uses scale.y
+      this._positionYIsUpToDate = false;
+    }
+
+    getDrawableX(): float {
+      if (this._object._absoluteCoordinates) {
+        return this._graphics.getLocalBounds().x;
+      }
+      let localBound = this._graphics.getLocalBounds().left;
```

Comment:
nitpicking: let's either use x or left, but be consistent to avoid confusion for the reader :) (same for y/top below)

Reply:
(unless they mean something different, but I don't think so?)

Reply:
Done (it was not marked as outdated)